File: Java/src/main/java/org/sisuns/bucles_while/intermedio/Ejercicio_03.java
package org.sisuns.bucles_while.intermedio;

public class Ejercicio_03 {
    public static void main(String[] args) {
        // Declaración de variables y asignación de valores
        String palabra1 = "stackoverflow";
        String palabra2 = "stackexchange";
        String prefijo = "";
        int longitud1 = palabra1.length();
        int longitud2 = palabra2.length();
        int i = 1;

        // Encontrar el prefijo común con un bucle while
        while (i <= Math.min(longitud1, longitud2)) {
            if (palabra1.substring(0, i).equals(palabra2.substring(0, i))) {
                prefijo = palabra1.substring(0, i);
            } else {
                break; // Si no hay prefijo común, salir del bucle
            }
            i++;
        }

        // Mostrar el resultado
        if (prefijo.length() > 0) {
            System.out.println("El prefijo común es: " + prefijo);
        } else {
            System.out.println("No existe un prefijo común.");
        }
    }
}

File: Python/Bucles_while/Intermedio/Ejercicio_03.py
# Lectura de las palabras desde la entrada estándar
palabra1 = input("Ingrese la primera palabra: ")
palabra2 = input("Ingrese la segunda palabra: ")

# Inicialización de variables
longitud1 = len(palabra1)
longitud2 = len(palabra2)
prefijo = ""
i = 1

# Encontrar el prefijo común con un bucle Mientras
while i <= min(longitud1, longitud2):
    if palabra1[:i] == palabra2[:i]:
        prefijo = palabra1[:i]
    else:
        break
    i += 1

# Mostrar el resultado
if len(prefijo) > 0:
    print("El prefijo común es:", prefijo)
else:
    print("No existe un prefijo común.")
